test/unit/tt_sigaction.c

```diff
@@ -0,0 +1,79 @@
+#include "unit.h"
+
+#include <pthread.h>
+#include <stdbool.h>
+#include <string.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include "tt_sigaction.h"
+#include "clock.h"
+
+/** Test duration in seconds. */
+#define TEST_LEN 1.5
+/** Signal sending period in usec. Must be lower than second. */
+#define SIGNAL_PERIOD 1e3
+
+pthread_t main_thread;
+
+void *
+thread_f(void *arg)
+{
+	double start = clock_monotonic();
+	while (clock_monotonic() - start < TEST_LEN)
+		sched_yield();
+	return NULL;
+}
+
+static void
+handler_f(int signum)
+{
+	(void)signum;
+	fail_if(!pthread_equal(pthread_self(), main_thread));
```

Comment:
Better count the number of times the signal handler was executed from a wrong thread and then check that it's zero in the end of the test.

Reply:
Done